### PR TITLE
Add sample application created with C#/.NET and Lambda

### PR DESCRIPTION
**Pull request by rohangupta**
*Description of changes:*
* Add sample application created with C#/.NET and Lambda
* Fix and add links to README.md

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dotnet/keyspaces-lambda-csharp/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+Then execute the test by using the .NET Core CLI
+
+```sh
+$ dotnet test test/keyspaces-lambda-csharp.Tests/
+```
+## Running the Sample Application
+### 0. Prerequisites
+There are some provided scripts for you to install, deploy and configure Lambda to your existing AWS account.
+
+Before running the scripts you need to load a profile file (or set environment variables) that grant access to AWS to create S3, IAM roles, and to setup Lambda functions.
```
Comment (bhouse99) [inline dotnet/keyspaces-lambda-csharp/README.md:53]: would be good to include some links here for credentials (aws profiles etc).

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+
+namespace keyspaces_lambda_csharp
+{
+    /*
+     * Represents the Lambda Function Handler.
+     * Returns "200 OK" if connection with Keyspaces is established successfully.
+     * Returns "400 Bad Request" in case of an error/exception.
+     */
+    public class Function
```
Comment (bhouse99) [inline dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs:18]: Call this something other than function... Lambda Sample, etc.  Function is a pretty overloaded term even with the namespace.

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                LambdaLogger.Log("CONTEXT: " + JsonConvert.SerializeObject(context));
+                LambdaLogger.Log("EVENT: " + JsonConvert.SerializeObject(invocationEvent));
+
+                int result = KeyspacesSession.Instance().validateConnection();
+                if (result == 200)
+                    return "200 OK";
+            }
+            catch (Exception e)
+            {
+                LambdaLogger.Log("Caught an exception: " + e.Message);
```
Comment (bhouse99) [inline dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs:34]: Probably a good idea to log more than the message.  Log the exception so you get a stack trace if something goes wrong.

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            }
+            return "400 Bad Request";
```
Comment (bhouse99) [inline dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs:36]: Include a log line that explains some of what went wrong (or at least that you didn't throw an exception etc).

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/KeyspacesSession.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        private void establishConnection()
+        {
+            if (session != null)
+                return;
+
+            string region = Environment.GetEnvironmentVariable("AWS_REGION");
+            string serviceUserName = Environment.GetEnvironmentVariable("SERVICE_USER_NAME");
+            string servicePassword = Environment.GetEnvironmentVariable("SERVICE_PASSWORD");
+            string certPath = Environment.GetEnvironmentVariable("CERTIFICATE_PATH");
+            LambdaLogger.Log("Region: " + region + ", Username: " + serviceUserName + ", Password: " + servicePassword + ", Certificate Path: " + certPath);
```
Comment (bhouse99) [inline dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/KeyspacesSession.cs:38]: Don't log the password.

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/KeyspacesSession.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        public int validateConnection()
+        {
+            if (session != null)
+            {
+                var rs = session.Execute("select * from system.local;");
+                var rows = rs.GetRows();
+                if (rows != null)
+                    return 200;
+            }
+            return 400;
```
Comment (bhouse99) [inline dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/KeyspacesSession.cs:65]: Include a lambda log about what went wrong (even if its just the deathrows of the session).

File: dotnet/keyspaces-lambda-csharp/test/keyspaces-lambda-csharp.Tests/FunctionTest.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace keyspaces_lambda_csharp.Tests
+{
+    /*
+     * Test class to validate the Lambda function handler locally.
+     * Run this to test the code before running any included scripts: 1-create-bucket.sh, 2-deploy.sh etc.
+     * 
+     * Set the environment variables (AWS_REGION, SERVICE_USER_NAME, SERVICE_PASSWORD and CERTIFICATE_PATH) before running this test.
+     * E.g.:
+     * AWS_REGION=us-west-2
```
Comment (bhouse99) [inline dotnet/keyspaces-lambda-csharp/test/keyspaces-lambda-csharp.Tests/FunctionTest.cs:15]: you should use the aws style examples.

**Review by bhouse99 — CHANGES_REQUESTED** [review]
Some comments and tweaks

File: README.md [hunk 8]
```diff
@@ -20,7 +20,8 @@ This is considered an intermediate learning resource and should typically be ref
 ## Examples <a name="Examples"></a>
 This repo contains examples in each language supported by Amazon Keyspaces. This is not exhaustive.
 
-- [Java Examples](https://github.com/aws-samples/aws-keyspaces-examples/tree/master/java)
+- [Java Examples](https://github.com/rohangupta/amazon-keyspaces-examples/tree/main/java)
```
Comment (bhouse99) [inline README.md:23]: Whops these point to your fork and not to the main repository

**Review by bhouse99 — CHANGES_REQUESTED** [review]

File: dotnet/keyspaces-lambda-csharp/README.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+```
+
+This setting enables the AWS CLI v2 to load JSON events from a file, matching the v1 behavior.
+
+## Running Tests
+
+### 0. Prerequisites
+
+Before submitting this job to Lambda it is recommended that you verify it locally on your developer machine.
```
[the remark this answers is not in the document]
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/README.md:33]: Yeah by "verifying" I just meant run the tests locally. 

Okay, I'll just omit this part.

File: dotnet/keyspaces-lambda-csharp/README.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+### 1. Environment Variables (Developer Machine)
+
+To connect to Amazon Keyspaces using Transport Layer Security (TLS), you need an Amazon digital certificate (`AmazonRootCA1.pem`) and configure the following environment variables:
```
[the remark this answers is not in the document]
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/README.md:37]: Sure, will provide the path for the cert.

File: dotnet/keyspaces-lambda-csharp/README.md (hunk 1 above)
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/README.md:53]: Done.

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs (hunk 2 above)
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs:18]: Renamed to LambdaSample.

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs (hunk 3 above)
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs:34]: Done.

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs (hunk 4 above)
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/Function.cs:36]: Added.

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/KeyspacesSession.cs (hunk 5 above)
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/KeyspacesSession.cs:38]: Removed.

File: dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/KeyspacesSession.cs (hunk 6 above)
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/src/keyspaces-lambda-csharp/KeyspacesSession.cs:65]: Added.

File: dotnet/keyspaces-lambda-csharp/test/keyspaces-lambda-csharp.Tests/FunctionTest.cs (hunk 7 above)
Comment (rohangupta) [reply dotnet/keyspaces-lambda-csharp/test/keyspaces-lambda-csharp.Tests/FunctionTest.cs:15]: Nice tip, modified.

File: README.md (hunk 8 above)
Comment (rohangupta) [reply README.md:23]: Oops, added the right links. Wanted to fix the original broken link.

**Review by bhouse99 — APPROVED** [review]

**Pull request merged**